### DISCUSSION TITLE
Livewire always sorts Collections

### DISCUSSION THREAD

**Discussion by francoism90** [livewire/livewire#6550] [7. Bugs (archived)]
### Livewire version

 v3.0.0

### Which PHP version are you using?

PHP 8.2

### Repository URL

https://github.com/filamentphp/filament

### Steps To Reproduce

See https://github.com/filamentphp/filament/discussions/7697 for details.

After some testing, it turns out Livewire always seems to sort Collection/Query results by key.
This causes things like search not to work (as the order is wrong). Not only in Filament, just linked to them, as it's easiest to test. It doesn't work in my own projects either.

It works fine outside Livewire, but when doing the same query in a Livewire Component, the orderByRaw doesn't seem to be called?

```php
Expected order: [11, 2, 3]
Result order: [2, 3, 11]
```

### Additional information

https://github.com/filamentphp/filament/discussions/7697

This wasn't an issue on Livewire V2. It seems the new Synth has something to do with this.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by PhiloNL** [comment]
Hi there! 
Would you be able to provide an entire code example from a component that replicates the bug or submit a failing test?
Thanks!

**Reply by francoism90** [reply]
@PhiloNL I don't know anymore.. it seems to be caused by something else.

It's weird, doing this (returns correct results):
```php
Book::search('a')->get()->pluck('id')
= Illuminate\Support\Collection {#8771
    all: [
      24627,
      27389,
      27390,
      27391,
      27392,
      27393,
      30432,
      8180,
    ],
  }
```
vs (implementing the search as seen by Filament)
```php
Book::query()->search('a')->pluck('id')
= Illuminate\Support\Collection {#8733
    all: [
      8180,
      24627,
      27389,
      27390,
      27391,
      27392,
      27393,
      30432,
      38188,
      38416,
    ],
  }
```
It seems like something is causing the order of keys to be sorted/reset, but I have no clue where.

Sorry, this isn't a Livewire issue, as I can produce the same results in Tinker.
